src/hotspot/share/oops/accessBackend.cpp

```diff
@@ -32,6 +32,12 @@
 #include "utilities/debug.hpp"
 #include "utilities/vmError.hpp"
 
+#if defined(TARGET_COMPILER_gcc)
+#define HIDDEN __attribute__ ((visibility ("hidden")))
```

Comment:
Couldn't this use [[gnu::visibility("hidden")]] instead, since HotSpot now allows C++14 Attributes? But this doesn't look like a very nice solution regardless

Reply:
We're currently using __attribute__ ((visibility ("X"))) elsewhere, so I think I should stick with that.

And, as I said in my long explanation, this is definitely a hack. The reason it is here is that I want to make sure the symbol table is 100% identical before and after this patch. 

Why this is needed, I don't know. I believe it is a bug in GCC, since we compile with -fvisibility=hidden. But I don't really know. Maybe there is something in the C++ standard that makes these functions behave differently. Perhaps they are even slightly incorrectly defined?

Reply:
Ah, I should have clarified that I did read the explaination, just that it slipped my mind at the time. I can't remember if C++14 or C++17 is the version for compilers ignoring unknown attributes though

Also, you can escape the underscores with \ to avoid GitHub formatting. GitHub can get annoying with its formatting sometimes

Is there a chance this can be fixed in a followup somehow? I could help with that, given I have access to WSL2 for Linux

Reply:
Oh no, by fix I meant removing the HIDDEN attribute and somehow fixing that issue differently, not replacing everything with C++ Attributes!

Reply:
Well, I hope that can be done, but it depends... As I see it, there are three possible ways to get rid of this `HIDDEN`.

1) Accept that we export these functions on gcc, and just remove the attribute. (Kind of capitulating to the bug.)

2) See if there is a way we can work around this problem, by changing something in how these functions are defined, that does not trigger this bug.

3) Have this bug fixed in gcc, and move to a minimum version of gcc that includes this fix.

I'm not terribly keen on any of these solutions. I think 2, if it is possible, would be the simplest and cleanest. Or 1, if we really think this HIDDEN workaround is worse than the problem it solves (unintentionally exporting symbols).

Reply:
This is not an issue with clang, only gcc.

My understanding was that the proper way to test for compiler in Hotspot code was by using `TARGET_COMPILER_gcc`.

I can try using the `#pragma` route instead; if it works I agree it is slightly better than to sprinkle `HIDDEN` all over the place.